src/io/image_io.cc

```diff
@@ -217,17 +217,28 @@ void Imread(const nnvm::NodeAttrs& attrs,
             std::vector<NDArray>* outputs) {
 #if MXNET_USE_OPENCV
   const auto& param = nnvm::get<ImreadParam>(attrs.parsed);
-
-  std::ifstream file(param.filename, std::ios::binary | std::ios::ate);
-  // if file is not open we get bad alloc after tellg
-  CHECK(file.is_open()) << "Imread: '" << param.filename
-      << "' couldn't open file: " << strerror(errno);
-  size_t fsize = file.tellg();
-  file.seekg(0, std::ios::beg);
+  dmlc::Stream *fi = dmlc::Stream::Create(param.filename.c_str(), "r", true);
+  CHECK(fi != NULL) << "Imread: '" << param.filename
+    << "' couldn't open file: " << strerror(errno);
+  std::vector<char*> buff_vec;
+  std::vector<int> buff_size;
+  int ptr = 0, single_buff_size = 10000;
```

Comment:
ptr you could declare in bottom for loop:
for(int i = 0, ptr = 0; ...
also ptr is kind of misleading, it’s more like an offset, right?

Reply:
songle_buf_size should be bigger unless there’s a good reason it’s so small. this could cause a lot of chatter for things like s3 which I assume sends a request for each read, and then all the backend churn that goes along server-side for that request.